### DISCUSSION TITLE
Add `expiresAt` to login information.

### DISCUSSION THREAD

**Discussion by patrick-emmanuel** [supabase/supabase#630] [General]
Hi,

thanks for the great product, I was wondering if it was possible to add `expiresAt` to the response payload for login. Reason is because I need this information to know when I need to request for a new token.

Also, an alternative would be how to get the signing key for the JWT. I can possibly decode the expiry time from that.

Thank you, please let me know if I need to clarify further!

**Comment by patrick-emmanuel** [comment]
So I found the JWT secret in Project > API Settings > JWT Secret!

**Reply by AndreaBarghigiani** [reply]
Hi @the-bionic I am not that comfortable with JWT nor authentication, would you be so kind to share a code sample to explain how would you check the expiration of the token?

I would like to do something similar in a side project I am creating. Basically I would like to alert the user that he needs to login before trying to send any data 😊

**Reply by patrick-emmanuel** [reply]
Hi @AndreaBarghigiani, sure, using axios interceptors and `dayjs` library. 

``` js
  axios.interceptors.request.use(async function (config) {
    // must've set token and expiresAt in localStorage after user logs in.
    const expireAt = localStorage.getItem('expiresAt');
    const refreshToken = localStorage.getItem('refreshToken');

    if (dayjs(expireAt).diff(dayjs()) < 1) {
      const { error, data } = await supabase.auth.api.refreshAccessToken(refreshToken);
      if (error != null) {
        return Promise.reject(error);
      }
      // save new token and refresh token to localStorage from data.
     
    }
    return config;
  }, function (error) {
    return Promise.reject(error);
  });
```
Ideally, you probably don't want your users to log in every time their token expires, you may use the refresh token to obtain a new valid token and they'll be authenticated. Axios interceptors run before every request checking token expiry.

To get the `expiresAt` value,  install `jsonwebtoken` and use:

`const decodedToken = jwt.verify(access_token, process.env.SUPABASE_JWT_SECRET);`

`decodedToken` should contain expiresAt value, that you should save to localStorage as well.
Hope this helps, please let me know if you need more clarification.
